### PR TITLE
Use correct query to get RAM Method for remote compilations

### PR DESCRIPTION
**Pull request by dchopra001**
Signed-off-by: Dhruv Chopra <Dhruv.C.Chopra@ibm.com>

### CODE REVIEW

File: runtime/compiler/z/codegen/S390J9CallSnippet.cpp [hunk 1]
```diff
@@ -357,7 +357,9 @@ TR::S390J9CallSnippet::emitSnippetBody()
          }
       else
          {
-         uintptrj_t ramMethod = (uintptr_t)methodSymbol->getMethodAddress();
+         // For remote compiles we do direct calls through snippets. However the method we're calling might already be compiled.
+         // In such a case we can't use getMethodAddress() to get the RAM Method because it returns the compiled code's entry point instead of the RAM Method.
+         uintptrj_t ramMethod = comp->getPersistentInfo()->getRemoteCompilationMode() == JITServer::SERVER && !methodSymbol->isInterpreted() ? (uintptrj_t)methodSymRef->getSymbol()->castToResolvedMethodSymbol()->getResolvedMethod()->getPersistentIdentifier() : (uintptrj_t)methodSymbol->getMethodAddress();
```
Comment (mpirvu) [inline runtime/compiler/z/codegen/S390J9CallSnippet.cpp]: Would it be wrong if we don't check whether the method is interpreted and always use `getPersistentIdentifier()`?
Also, to identify server size compilations `comp->isOutOfProcessCompilation()` is preferable.

Comment (dchopra001) [reply runtime/compiler/z/codegen/S390J9CallSnippet.cpp]: Just to clarify, are you proposing the following change?
```
uintptrj_t ramMethod = ..getPersistentIdentifier();
```

Or something like this:
```
uintptrj_t ramMethod = comp->isOutOfProcessCompilation() ? ..getPersistentIdentifier() : (uintptrj_t)methodSymbol->getMethodAddress();
```

Comment (mpirvu) [reply runtime/compiler/z/codegen/S390J9CallSnippet.cpp]: I was suggesting the latter, but now that you mentioned it, why is the first variation wrong? If all we want is the j9method, then getPersistentIdentifier() will us that

Comment (dchopra001) [reply runtime/compiler/z/codegen/S390J9CallSnippet.cpp]: Yes, that's why I was clarifying. I think `getPersistentIdentifier()` by itself should be sufficient. However I'd like to test it on JITServer as well as master. I'll report back once I have the results. If good, then we can make this change directly in the master branch.

**Comment by dchopra001** [comment]
Currently testing this in the master branch as well as my jitserver branch.

**Comment by dchopra001** [comment]
@mpirvu Both builds passed. I have repurposed this PR to target the master branch. This should be good to merge now.

**Comment by mpirvu** [comment]
Jenkins test sanity zlinux jdk8,jdk11

**Comment by mpirvu** [comment]
Tests have passed, hence merging.

**Pull request merged**